ipfx/bin/run_feature_vector_extraction.py

```diff
@@ -19,20 +19,67 @@ class CollectFeatureVectorParameters(ags.ArgSchema):
     output_dir = ags.fields.OutputDir(default=None)
     input = ags.fields.InputFile(default=None, allow_none=True)
     project = ags.fields.String(default="T301")
-    include_failed_sweeps = ags.fields.Boolean(default=False)
+    sweep_qc_option = ags.fields.String(default=None, allow_none=True)
     include_failed_cells = ags.fields.Boolean(default=False)
     run_parallel = ags.fields.Boolean(default=True)
     ap_window_length = ags.fields.Float(description="Duration after threshold for AP shape (s)", default=0.003)
 
 
-def categorize_iclamp_sweeps(data_set, stimuli_names, passed_only=False):
+def categorize_iclamp_sweeps(data_set, stimuli_names, sweep_qc_option=None, specimen_id=None):
     # TODO - deal with pass/fail status
 
-    iclamp_st = data_set.filtered_sweep_table(clamp_mode=data_set.CURRENT_CLAMP, stimuli=stimuli_names)
-    return iclamp_st["sweep_number"].sort_values().values
+    passed_sql = """
+    select swp.sweep_number from ephys_sweeps swp
+    where swp.specimen_id = %s
+    and swp.sweep_number = any(%s)
+    and swp.workflow_state like '%%passed'
+    """
 
+    iclamp_st = data_set.filtered_sweep_table(clamp_mode=data_set.CURRENT_CLAMP, stimuli=stimuli_names)
 
-def data_for_specimen_id(specimen_id, passed_only, ap_window_length=0.005):
```

Comment:
I wish we did not need to query LIMS here, since this prevents using this functionality by external users. We should be able to modify the QC module to allow selecting the desired sweeps according to the `sweep_qc_option`. We should strive to limit usage of `specimen_id` to only generate top level input (i.e., nwb_file, h5_file), otherwise external users would not be able to use it.

Reply:
I don’t think I necessarily agree with the broader comment - it’s a runner script, not a library. So some level of specificity for the interface is necessary here.

More specifically, there are a host of issues with the QC status of sweeps in the NWB files that are actually being used, so it’s just not possible to rely on that directly at this time.

I could take the runner scripts out of the library entirely, but it seems like there is some use of having them as examples to learn from.